src/swarm-lang/Swarm/Language/Pipeline/Cmdline.hs

```diff
@@ -0,0 +1,29 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+-- |
+-- SPDX-License-Identifier: BSD-3-Clause
+--
+-- Parsing + typechecking swarm-lang code at the command line.
+module Swarm.Language.Pipeline.Cmdline where
+
+import Data.Text.IO qualified as T
+import Swarm.Language.Pipeline (processTerm)
+import Swarm.Util.InputSource
+import System.Exit (exitFailure)
+import System.IO (stderr)
+
+newtype CheckConfig = CheckConfig {checkInput :: InputSource}
+
+-- | Validate swarm-lang code.
+checkSwarmIO :: CheckConfig -> IO ()
+checkSwarmIO (CheckConfig input) = do
+  T.putStrLn $ "Checking " <> showInput input <> " ..."
```

Comment:
It's a bit of work, but it would be nice if this was just `putStr` and the OK got put on the same line.